crates/mun_hir/src/ty/infer.rs

```diff
@@ -400,9 +400,23 @@ impl<'a, D: HirDatabase> InferenceResultBuilder<'a, D> {
                     }
                 }
             }
-            Expr::UnaryOp { .. } => Ty::Unknown,
-            //            Expr::UnaryOp { expr: _, op: _ } => {}
-            //            Expr::Block { statements: _, tail: _ } => {}
+            Expr::UnaryOp { expr, op } => {
+                let ty = self.infer_expr(*expr, &Expectation::none());
+                if let Some(simple) = ty.as_simple() {
+                    match op {
+                        UnaryOp::Not => match simple {
+                            TypeCtor::Bool | TypeCtor::Int(_) => ty,
+                            _ => Ty::Unknown,
```

Comment:
There should be an error emitted here. Something similar to binary operation error: ```cannot apply operator `-` on type `int`.```

Reply:
So a panic?

Reply:
See line 308 of this same file. There is a `Diagnostic` added: `CannotApplyBinaryOp`. There should be something similar with a unary operator.